### PR TITLE
Feature/grid part4

### PR DESCRIPTION
**Pull request by romaninsh**
Hopefully last PR in Grid implementation.

 - added paginator
 - added documentation
 - cleaned up code
 - added comments

screen shot 2017-03-31 at 14 49 49

### CODE REVIEW

File: src/Grid.php [hunk 1]
```diff
@@ -51,8 +92,27 @@ public function addQuickSearch($fields = [])
...
+        $this->quickSearch = $form->add(new \atk4\ui\FormField\Input(['placeholder'=>'Search', 'short_name'=>$this->name.'_q', 'icon'=>'search']))
+            ->addClass('transparent');
+
+        if (isset($_GET[$this->name.'_q'])) {
+            $q = $_GET[$this->name.'_q'];
+            $this->quickSearch->set($q);
+
+            $cond = [];
+            foreach ($fields as $field) {
+                $cond[] = [$field, 'like', '%'.$q.'%'];
```
Comment (DarkSide666) [inline src/Grid.php]: Is this safe for SQL injections?

File: src/Paginator.php [hunk 2]
```diff
@@ -57,6 +47,20 @@ public function init()
     }
 
     /**
+     * Set total number of pages.
+     */
+    public function setTotal($total)
+    {
+        $this->total = (int) $total;
+
+        if ($this->page < 1) {
```
Comment (DarkSide666) [inline src/Paginator.php]: `$this->page < 1` is not needed here. Only `> $this->total` is needed.

File: src/Grid.php (hunk 1 above)
Comment (romaninsh) [inline src/Grid.php]: yep. 3rd argument is escaped.

File: src/Paginator.php (hunk 2 above)
Comment (romaninsh) [inline src/Paginator.php]: if someone would manually put page = 0, we don't any weird stuff.

**Pull request merged**